### DISCUSSION TITLE
Deploment 404 on Netlify

### DISCUSSION THREAD

**Discussion by sergioloporto** [vercel/next.js#19868] [Help]
Hello, I am trying to deploy a site with Netlify. [This](https://github.com/sergioloporto/BookLanding) is the repo. [Here](https://booklanding.netlify.app/) is the link to the deployed site.

Here is the deploy log:

```
1:37:18 PM: Build ready to start
1:37:20 PM: build-image version: 09c2cdcdf242cf2f57c9ee0fcad9d298fad9ad41
1:37:20 PM: build-image tag: v3.5.0
1:37:20 PM: buildbot version: 5988d267666895e50a3b61cb809f46c82b965ece
1:37:20 PM: Fetching cached dependencies
1:37:20 PM: Starting to download cache of 144.7MB
1:37:21 PM: Finished downloading cache in 854.494405ms
1:37:21 PM: Starting to extract cache
1:37:25 PM: Finished extracting cache in 4.194208368s
1:37:25 PM: Finished fetching cache in 5.102512401s
1:37:25 PM: Starting to prepare the repo for build
1:37:25 PM: Preparing Git Reference refs/heads/master
1:37:28 PM: Starting build script
1:37:28 PM: Installing dependencies
1:37:28 PM: Python version set to 2.7
1:37:29 PM: Started restoring cached node version
1:37:31 PM: Finished restoring cached node version
1:37:32 PM: v12.18.0 is already installed.
1:37:33 PM: Now using node v12.18.0 (npm v6.14.4)
1:37:33 PM: Started restoring cached build plugins
1:37:33 PM: Finished restoring cached build plugins
1:37:33 PM: Attempting ruby version 2.7.1, read from environment
1:37:34 PM: Using ruby version 2.7.1
1:37:34 PM: Using PHP version 5.6
1:37:34 PM: Started restoring cached node modules
1:37:34 PM: Finished restoring cached node modules
1:37:35 PM: Installing NPM modules using NPM version 6.14.4
1:37:39 PM: npm WARN react-no-ssr@1.1.0 requires a peer of react@^0.14.0 || ^15.0.0 but none is installed. You must install peer dependencies yourself.
1:37:39 PM: npm WARN mojosa@2.8.0 No repository field.
1:37:39 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
1:37:39 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
1:37:39 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
1:37:39 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
1:37:39 PM: audited 727 packages in 3.893s
1:37:40 PM: 32 packages are looking for funding
1:37:40 PM:   run `npm fund` for details
1:37:40 PM: found 3 low severity vulnerabilities
1:37:40 PM:   run `npm audit fix` to fix them, or `npm audit` for details
1:37:40 PM: NPM modules installed
1:37:40 PM: Started restoring cached go cache
1:37:40 PM: Finished restoring cached go cache
1:37:40 PM: go version go1.14.4 linux/amd64
1:37:40 PM: go version go1.14.4 linux/amd64
1:37:40 PM: Installing missing commands
1:37:40 PM: Verify run directory
1:37:42 PM: ​
1:37:42 PM: ────────────────────────────────────────────────────────────────
1:37:42 PM:   Netlify Build                                                 
1:37:42 PM: ────────────────────────────────────────────────────────────────
1:37:42 PM: ​
1:37:42 PM: ❯ Version
1:37:42 PM:   @netlify/build 5.5.5
1:37:42 PM: ​
1:37:42 PM: ❯ Flags
1:37:42 PM:   deployId: 5fcb7efee65e3a0007ff7a9e
1:37:42 PM:   mode: buildbot
1:37:42 PM: ​
1:37:42 PM: ❯ Current directory
1:37:42 PM:   /opt/build/repo
1:37:42 PM: ​
1:37:42 PM: ❯ Config file
1:37:42 PM:   No config file was defined: using default values.
1:37:42 PM: ​
1:37:42 PM: ❯ Context
1:37:42 PM:   production
1:37:42 PM: ​
1:37:42 PM: ────────────────────────────────────────────────────────────────
1:37:42 PM:   1. Build command from Netlify app                             
1:37:42 PM: ────────────────────────────────────────────────────────────────
1:37:42 PM: ​
1:37:42 PM: $ npm run build
1:37:42 PM: > mojosa@2.8.0 build /opt/build/repo
1:37:42 PM: > next build
1:37:43 PM: info  - Creating an optimized production build...
1:38:33 PM: info  - Compiled successfully
1:38:33 PM: info  - Collecting page data...
1:38:33 PM: Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
1:38:33 PM: Read more: https://err.sh/next.js/opt-out-auto-static-optimization
1:38:33 PM: Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
1:38:33 PM: Read more: https://err.sh/next.js/opt-out-auto-static-optimization
1:38:34 PM: Page                                                           Size     First Load JS
1:38:34 PM: ┌ λ /                                                          155 B           103 kB
1:38:34 PM: ├   /_app                                                      0 B            75.2 kB
1:38:34 PM: ├ λ /404                                                       2.69 kB        77.9 kB
1:38:34 PM: ├ λ /about-one                                                 3.19 kB         106 kB
1:38:34 PM: ├ λ /about-two                                                 3.44 kB         107 kB
1:38:34 PM: ├ λ /agency                                                    9.76 kB         119 kB
1:38:34 PM: ├ λ /app                                                       6.4 kB          116 kB
1:38:34 PM: ├ λ /blog-details                                              2.3 kB          106 kB
1:38:34 PM: ├ λ /blog-one                                                  1.41 kB         105 kB
1:38:34 PM: ├ λ /blog-two                                                  1.82 kB         105 kB
1:38:34 PM: ├ λ /cart                                                      1.92 kB         105 kB
1:38:34 PM: ├ λ /checkout                                                  11.8 kB         115 kB
1:38:34 PM: ├ λ /coming-soon                                               1.77 kB          77 kB
1:38:34 PM: ├ λ /contact                                                   1.48 kB         105 kB
1:38:34 PM: ├ λ /digital-marketing                                         9.31 kB         119 kB
1:38:34 PM: ├ λ /faq                                                       4.78 kB         108 kB
1:38:34 PM: ├ λ /features                                                  2.09 kB         105 kB
1:38:34 PM: ├ λ /gallery                                                   1.78 kB         105 kB
1:38:34 PM: ├ λ /login                                                     1.17 kB         104 kB
1:38:34 PM: ├ λ /payment-processing                                        7.18 kB         110 kB
1:38:34 PM: ├ λ /portfolio-one                                             1.46 kB         105 kB
1:38:34 PM: ├ λ /portfolio-three                                           1.48 kB         105 kB
1:38:34 PM: ├ λ /portfolio-two                                             1.47 kB         105 kB
1:38:34 PM: ├ λ /pricing-one                                               1.22 kB         104 kB
1:38:34 PM: ├ λ /pricing-three                                             1.11 kB         104 kB
1:38:34 PM: ├ λ /pricing-two                                               1.11 kB         104 kB
1:38:34 PM: ├ λ /product-landing                                           6.48 kB         110 kB
1:38:34 PM: ├ λ /project-details                                           1.97 kB         105 kB
1:38:34 PM: ├ λ /saas                                                      8.38 kB         118 kB
1:38:34 PM: ├ λ /services                                                  2.35 kB         106 kB
1:38:34 PM: ├ λ /shop-details                                              3.08 kB         106 kB
1:38:34 PM: ├ λ /shop-one                                                  1.83 kB         105 kB
1:38:34 PM: ├ λ /shop-two                                                  2.27 kB         106 kB
1:38:34 PM: ├ λ /signup                                                    1.14 kB         104 kB
1:38:34 PM: ├ λ /software-landing                                          5.51 kB         115 kB
1:38:34 PM: ├ λ /team-one                                                  1.11 kB         104 kB
1:38:34 PM: ├ λ /team-two                                                  1.14 kB         104 kB
1:38:34 PM: └ λ /thankyou                                                  807 B           104 kB
1:38:34 PM: + First Load JS shared by all                                  75.2 kB
1:38:34 PM:   ├ chunks/6e82a459cbd936bb22ee277ca91cac38288c570d.e1f4e8.js  12.8 kB
1:38:34 PM:   ├ chunks/b8612b88f7d3fea52646b544ce4fbd88373b49ea.e69209.js  7.03 kB
1:38:34 PM:   ├ chunks/framework.9e938b.js                                 40.4 kB
1:38:34 PM:   ├ chunks/main.bce543.js                                      6.54 kB
1:38:34 PM:   ├ chunks/pages/_app.2e4503.js                                7.29 kB
1:38:34 PM:   ├ chunks/webpack.3b269c.js                                   1.18 kB
1:38:34 PM:   └ css/bcf4d7e39a2d48e5cb09.css                               60.9 kB
1:38:34 PM: λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
1:38:34 PM: ○  (Static)  automatically rendered as static HTML (uses no initial props)
1:38:34 PM: ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
1:38:34 PM:    (ISR)     incremental static regeneration (uses revalidate in getStaticProps)
1:38:34 PM: ​
1:38:34 PM: (build.command completed in 52.5s)
1:38:34 PM: ​
1:38:34 PM: ────────────────────────────────────────────────────────────────
1:38:34 PM:   Netlify Build Complete                                        
1:38:34 PM: ────────────────────────────────────────────────────────────────
1:38:34 PM: ​
1:38:34 PM: (Netlify Build completed in 52.6s)
1:38:34 PM: Caching artifacts
1:38:34 PM: Started saving node modules
1:38:34 PM: Finished saving node modules
1:38:34 PM: Started saving build plugins
1:38:34 PM: Finished saving build plugins
1:38:34 PM: Started saving pip cache
1:38:34 PM: Finished saving pip cache
1:38:34 PM: Started saving emacs cask dependencies
1:38:34 PM: Finished saving emacs cask dependencies
1:38:34 PM: Started saving maven dependencies
1:38:34 PM: Finished saving maven dependencies
1:38:34 PM: Started saving boot dependencies
1:38:34 PM: Finished saving boot dependencies
1:38:34 PM: Started saving go dependencies
1:38:34 PM: Finished saving go dependencies
1:38:34 PM: Build script success
1:38:34 PM: Starting to deploy site from ''
1:38:35 PM: Creating deploy tree asynchronously
1:38:35 PM: Creating deploy upload records
1:38:37 PM: 1 new files to upload
1:38:37 PM: 0 new functions to upload
1:38:37 PM: Starting post processing
1:38:37 PM: Post processing - HTML
1:38:38 PM: Post processing - header rules
1:38:38 PM: Post processing - redirect rules
1:38:38 PM: Post processing done
1:38:38 PM: Site is live ✨
1:38:56 PM: Finished processing build request in 1m36.271583443s
```

Here is the deployment configuration:

```
Base directory - Not set
Build command- npm run build
Publish directory - Not set
Deploy log visibility - Logs are public
Builds - Active
```

I didn't set a `Publish directory` because it's overwritten by the script to `/.next` and this is the right folder.

Can anyone spot what I am doing wrong and what I have to change to fix it?

**Comment by Nandhini-Madan** [comment]
I having the same issue and I don't know where I am going wrong

12:21:20 PM: Build ready to start
12:21:21 PM: build-image version: 0e2f4c52031ab562db66aec633308326e3b108d0 (focal)
12:21:21 PM: build-image tag: focal
12:21:21 PM: buildbot version: 0d1cf41e4e345a224c39ab2a783aa4af5c76f6e8
12:21:21 PM: Fetching cached dependencies
12:21:21 PM: Starting to download cache of 154.2MB
12:21:23 PM: Finished downloading cache in 1.164594691s
12:21:23 PM: Starting to extract cache
12:21:26 PM: Finished extracting cache in 3.403708704s
12:21:26 PM: Finished fetching cache in 4.641024598s
12:21:26 PM: Starting to prepare the repo for build
12:21:26 PM: Preparing Git Reference refs/heads/main
12:21:26 PM: Parsing package.json dependencies
12:21:27 PM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'nwnext' versus '' in the Netlify UI
12:21:28 PM: Starting build script
12:21:28 PM: Installing dependencies
12:21:28 PM: Python version set to 2.7
12:21:28 PM: Started restoring cached node version
12:21:29 PM: Finished restoring cached node version
12:21:30 PM: v16.16.0 is already installed.
12:21:30 PM: Now using node v16.16.0 (npm v8.11.0)
12:21:30 PM: Started restoring cached build plugins
12:21:30 PM: Finished restoring cached build plugins
12:21:30 PM: Attempting ruby version 2.7.2, read from environment
12:21:31 PM: Using ruby version 2.7.2
12:21:32 PM: Using PHP version 8.0
12:21:32 PM: No npm workspaces detected
12:21:32 PM: Started restoring cached node modules
12:21:32 PM: Finished restoring cached node modules
12:21:32 PM: Started restoring cached go cache
12:21:32 PM: Finished restoring cached go cache
12:21:32 PM: Installing Go version 1.17 (requested 1.17)
12:21:38 PM: unset GOOS;
12:21:38 PM: unset GOARCH;
12:21:38 PM: export GOROOT='/opt/buildhome/.gimme/versions/go1.17.linux.amd64';
12:21:38 PM: export PATH="/opt/buildhome/.gimme/versions/go1.17.linux.amd64/bin:${PATH}";
12:21:38 PM: go version >&2;
12:21:38 PM: export GIMME_ENV="/opt/buildhome/.gimme/env/go1.17.linux.amd64.env"
12:21:38 PM: go version go1.17 linux/amd64
12:21:38 PM: Installing missing commands
12:21:38 PM: Verify run directory
12:21:39 PM: ​
12:21:39 PM: ────────────────────────────────────────────────────────────────
12:21:39 PM:   Netlify Build                                                 
12:21:39 PM: ────────────────────────────────────────────────────────────────
12:21:39 PM: ​
12:21:39 PM: ❯ Version
12:21:39 PM:   @netlify/build 27.3.4
12:21:39 PM: ​
12:21:39 PM: ❯ Flags
12:21:39 PM:   baseRelDir: true
12:21:39 PM:   buildId: 62d1be2f678e376d67ca7bb5
12:21:39 PM:   deployId: 62d1be2f678e376d67ca7bb7
12:21:39 PM: ​
12:21:39 PM: ❯ Current directory
12:21:39 PM:   /opt/build/repo/nwnext
12:21:39 PM: ​
12:21:39 PM: ❯ Config file
12:21:39 PM:   No config file was defined: using default values.
12:21:39 PM: ​
12:21:39 PM: ❯ Context
12:21:39 PM:   production
12:21:40 PM: ​
12:21:40 PM: ────────────────────────────────────────────────────────────────
12:21:40 PM:   1. Build command from Netlify app                             
12:21:40 PM: ────────────────────────────────────────────────────────────────
12:21:40 PM: ​
12:21:40 PM: $ next build
12:21:40 PM: info  - SWC minify release candidate enabled. https://nextjs.link/swcmin
12:21:40 PM: warn  - No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
12:21:40 PM: info  - Linting and checking validity of types...
12:21:42 PM: info  - Creating an optimized production build...
12:21:46 PM: info  - Compiled successfully
12:21:46 PM: info  - Collecting page data...
12:21:50 PM: info  - Generating static pages (0/4)
12:21:50 PM: info  - Generating static pages (1/4)
12:21:51 PM: info  - Generating static pages (2/4)
12:21:51 PM: info  - Generating static pages (3/4)
12:21:51 PM: info  - Generating static pages (4/4)
12:21:51 PM: info  - Finalizing page optimization...
12:21:51 PM: Page                                       Size     First Load JS
12:21:51 PM: ┌ ○ /                                      5.81 kB        83.9 kB
12:21:51 PM: ├   └ css/149b18973e5508c7.css             655 B
12:21:51 PM: ├   /_app                                  0 B            78.1 kB
12:21:51 PM: ├ ○ /404                                   186 B          78.3 kB
12:21:51 PM: Creating deploy upload records
12:21:51 PM: ├ λ /api/hello                             0 B            78.1 kB
12:21:51 PM: └ ○ /post/first                            2.45 kB        80.6 kB
12:21:51 PM: + First Load JS shared by all              78.1 kB
12:21:51 PM:   ├ chunks/framework-ae4f43955bfa5ddc.js   45.7 kB
12:21:51 PM:   ├ chunks/main-3cf1b4d0a841b0b4.js        31.1 kB
12:21:51 PM:   ├ chunks/pages/_app-6e0f6f809c4ba91c.js  509 B
12:21:51 PM:   ├ chunks/webpack-34c76af51fb9f015.js     760 B
12:21:51 PM:   └ css/27d177a30947857b.css               194 B
12:21:51 PM: λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
12:21:51 PM: ○  (Static)  automatically rendered as static HTML (uses no initial props)
12:21:51 PM: ​
12:21:51 PM: Starting post processing
12:21:51 PM: (build.command completed in 11.3s)
12:21:51 PM: ​
12:21:51 PM: ────────────────────────────────────────────────────────────────
12:21:51 PM:   2. Deploy site                                                
12:21:51 PM: Post processing - HTML
12:21:51 PM: ────────────────────────────────────────────────────────────────
12:21:51 PM: ​
12:21:52 PM: Post processing - header rules
12:21:51 PM: Starting to deploy site from 'nwnext'
12:21:51 PM: Creating deploy tree 
12:21:51 PM: 1 new files to upload
12:21:52 PM: Post processing - redirect rules
12:21:51 PM: 0 new functions to upload
12:21:51 PM: Site deploy was successfully initiated
12:21:51 PM: ​
12:21:52 PM: Post processing done
12:21:51 PM: (Deploy site completed in 477ms)
12:21:51 PM: ​
12:21:51 PM: ────────────────────────────────────────────────────────────────
12:21:51 PM:   Netlify Build Complete                                        
12:21:51 PM: ────────────────────────────────────────────────────────────────
12:21:51 PM: ​
12:21:51 PM: (Netlify Build completed in 11.8s)
12:21:52 PM: Caching artifacts
12:21:52 PM: Started saving node modules
12:21:52 PM: Finished saving node modules
12:21:52 PM: Started saving build plugins
12:21:52 PM: Finished saving build plugins
12:21:52 PM: Started saving pip cache
12:21:52 PM: Finished saving pip cache
12:21:52 PM: Started saving emacs cask dependencies
12:21:52 PM: Finished saving emacs cask dependencies
12:21:52 PM: Started saving maven dependencies
12:21:52 PM: Finished saving maven dependencies
12:21:52 PM: Started saving boot dependencies
12:21:52 PM: Finished saving boot dependencies
12:21:52 PM: Started saving rust rustup cache
12:21:52 PM: Finished saving rust rustup cache
12:21:52 PM: Started saving go dependencies
12:21:52 PM: Finished saving go dependencies
12:21:52 PM: Build script success
12:21:54 PM: Uploading Cache of size 154.0MB
12:21:55 PM: Site is live ✨
12:21:57 PM: Finished processing build request in 36.027940402s
